### PR TITLE
增加dijkstra算法相关内容及相关代码示例

### PR DESCRIPTION
**Pull request by Aitensa**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by nuomi1** [comment]
其他语言的代码都是自己实现的吗，还是借助 AI 生成的呀？

**Comment by Aitensa** [comment]
是借助AI进行了修饰，经过测试环境运行

File: codes/c/chapter_greedy/dijkstra.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <limits.h>
```
Comment (Gonglja) [inline codes/c/chapter_greedy/dijkstra.c:3]: 将头文件添加至公共头文件中，仅需包含一个 `#include "../utils/common.h"`

Comment (Gonglja) [inline codes/c/chapter_greedy/dijkstra.c:3]: 将头文件添加至公共头文件中，仅需包含一个 #include "../utils/common.h"

Comment (Gonglja) [reply codes/c/chapter_greedy/dijkstra.c:3]: 另外添加文件信息，
比如
```cpp
/**
 * File: n_queens.cpp
 * Created Time: 2023-05-04
 * Author: Krahets (krahets@163.com)
 */
```

File: codes/cpp/chapter_greedy/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 add_executable(coin_change_greedy coin_change_greedy.cpp)
 add_executable(fractional_knapsack fractional_knapsack.cpp)
-add_executable(max_capacity max_capacity.cpp)
\ No newline at end of file
+add_executable(max_capacity max_capacity.cpp)
+add_executable(Dijkstra Dijkstra.cpp)
```
Comment (Gonglja) [inline codes/cpp/chapter_greedy/CMakeLists.txt:4]: 小写

File: codes/cpp/chapter_greedy/Dijkstra.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+#include <queue>
+#include <vector>
+#include <limits.h>
```
Comment (Gonglja) [inline codes/cpp/chapter_greedy/Dijkstra.cpp:4]: 同上

**Review by Gonglja — COMMENTED** [review]
看了下 c 和 cpp 的代码，有点问题标出来了
其它的需要 @krahets 和其他大佬来继续看喽

**Pull request closed without merging**